test/ConstructiveSolidGeometry/CSG_primitives.jl

```diff
@@ -79,7 +79,12 @@ no_translations = (rotation = one(SMatrix{3, 3, T, 9}), translation = zero(Carte
         @inferred CSG.Cone{Float64}(r=((3,5),nothing))
         @inferred CSG.Cone{Float64}(r=(1,2))
         @inferred CSG.Cone{Float64}(r=(nothing,(3,5)))
-        @inferred CSG.Cone{Float64}(r=((1),(2)))
+        @inferred CSG.Cone{Float64}(r=((1,),(2,)))
+        @inferred CSG.Cone(r=(nothing,(3,5)))
+        @inferred CSG.Cone(r=((3,5),nothing))
+        @inferred CSG.Cone(r=(1,2))
+        @inferred CSG.Cone(r=(nothing,(3,5)))
+        @inferred CSG.Cone(r=((1,),(2,)))
```

Comment:
Looks like this still needs another function to get the promoted CSG type:

```julia
_csg_get_promoted_eltype(::Type{Tuple{T}}) where {T<:Real} = T
```
?